webapp/src/shared/components/Text/Text.tsx

```diff
@@ -32,17 +33,13 @@ type TextProps = {
   children: string;
 };
 
-export default function Text({
-  variant,
-  color = TextColor.DARK,
-  weight = TextWeight.REGULAR,
-  children,
-}: TextProps) {
-  const { tag, size } = variant;
+export default function Text({ variant, color, weight, children }: TextProps) {
+  const { tag, className } = variant;
   const TextTag = tag as React.ElementType;
+  const textClassName = `${className} ${color ? color : ''}`;
 
   return (
-    <TextTag style={{ fontSize: size, fontWeight: weight }} className={color}>
+    <TextTag style={{ fontWeight: weight }} className={textClassName}>
```

Comment:
if weigth is undefined, what will this expand to?
I think it would be
```font-weight: undefined;```

Reply:
I think if the weight is undefined, React does not set an inline `font-weight` for this element.